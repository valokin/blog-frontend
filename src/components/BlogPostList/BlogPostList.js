import { useHistory } from "react-router-dom";
import InfiniteScroll from "react-infinite-scroll-component";

import BlogPostListItem from '../BlogPostListItem/BlogPostListItem';

const BlogPostList = ({ postsWithComments, fetchMoreComments }) => {

  const history = useHistory();
  return (
    <div>
      <h1>Blog</h1>
      <InfiniteScroll
        dataLength={postsWithComments.length}
        next={fetchMoreComments}
        hasMore={true}
        loader={<h4>Loading...</h4>}
      >
        {postsWithComments.map((post) => {
          return (
            <BlogPostListItem
              onClick={() => {
                history.push(`/post/${post.id}`);
              }}
              post={post}
              key={post.id}>

            </BlogPostListItem>
          )
        })}
      </InfiniteScroll>
    </div>
  )
};

export default BlogPostList;
