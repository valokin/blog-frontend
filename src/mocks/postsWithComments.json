[
    {
        "id": "1",
        "createdAt": 1598278023,
        "title": "title 1",
        "description": "description 1",
        "text": "text 1",
        "comments": [
            {
                "id": "1",
                "postId": "1",
                "name": "Austin Kuvalis",
                "text": "text 1",
                "createdAt": 1598277994
            },
            {
                "id": "10",
                "postId": "1",
                "name": "Craig O'Kon",
                "text": "text 10",
                "createdAt": 1598277459
            },
            {
                "id": "17",
                "postId": "1",
                "name": "Jake Prosacco",
                "text": "text 17",
                "createdAt": 1598277093
            }
        ]
    },
    {
        "id": "2",
        "createdAt": 1598277963,
        "title": "title 2",
        "description": "description 2",
        "text": "text 2",
        "comments": [
            {
                "id": "2",
                "postId": "2",
                "name": "Travis Shanahan",
                "text": "text 2",
                "createdAt": 1598277934
            },
            {
                "id": "11",
                "postId": "2",
                "name": "Saul Hettinger",
                "text": "text 11",
                "createdAt": 1598277399
            },
            {
                "id": "18",
                "postId": "2",
                "name": "Coty Wilderman",
                "text": "text 18",
                "createdAt": 1598277033
            }
        ]
    },
    {
        "id": "3",
        "createdAt": 1598277903,
        "title": "title 3",
        "description": "description 3",
        "text": "text 3",
        "comments": [
            {
                "id": "3",
                "postId": "3",
                "name": "Brice Lang",
                "text": "text 3",
                "createdAt": 1598277874
            },
            {
                "id": "12",
                "postId": "3",
                "name": "Keegan Bradtke",
                "text": "text 12",
                "createdAt": 1598277339
            },
            {
                "id": "19",
                "postId": "3",
                "name": "Thalia Reichel",
                "text": "text 19",
                "createdAt": 1598276973
            }
        ]
    },
    {
        "id": "4",
        "createdAt": 1598277843,
        "title": "title 4",
        "description": "description 4",
        "text": "text 4",
        "comments": [
            {
                "id": "4",
                "postId": "4",
                "name": "Doug Sawayn",
                "text": "text 4",
                "createdAt": 1598277814
            },
            {
                "id": "13",
                "postId": "4",
                "name": "Rory Towne",
                "text": "text 13",
                "createdAt": 1598277279
            },
            {
                "id": "20",
                "postId": "4",
                "name": "Clyde Lind",
                "text": "text 20",
                "createdAt": 1598276913
            }
        ]
    },
    {
        "id": "5",
        "createdAt": 1598277783,
        "title": "title 5",
        "description": "description 5",
        "text": "text 5",
        "comments": [
            {
                "id": "5",
                "postId": "5",
                "name": "Oran Kovacek",
                "text": "text 5",
                "createdAt": 1598277754
            },
            {
                "id": "14",
                "postId": "5",
                "name": "Abigayle Schinner V",
                "text": "text 14",
                "createdAt": 1598277219
            },
            {
                "id": "21",
                "postId": "5",
                "name": "Sydnee Lockman",
                "text": "text 21",
                "createdAt": 1598276853
            }
        ]
    },
    {
        "id": "6",
        "createdAt": 1598277723,
        "title": "title 6",
        "description": "description 6",
        "text": "text 6",
        "comments": [
            {
                "id": "6",
                "postId": "6",
                "name": "Bonnie Beer",
                "text": "text 6",
                "createdAt": 1598277694
            },
            {
                "id": "15",
                "postId": "6",
                "name": "Leanna Boyle",
                "text": "text 15",
                "createdAt": 1598277159
            },
            {
                "id": "22",
                "postId": "6",
                "name": "Jaime Champlin",
                "text": "text 22",
                "createdAt": 1598276793
            }
        ]
    }
]