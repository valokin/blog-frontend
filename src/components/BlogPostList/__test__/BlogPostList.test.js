import React from 'react';
import ReactDom from 'react-dom';
import BlogPostList from '../BlogPostList';
import postsWithComments from '../../../mocks/postsWithComments.json';

import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

const fetchMoreComments = () => {
}

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(<BlogPostList 
        postsWithComments={postsWithComments}
        fetchMoreComments={fetchMoreComments}
    ></BlogPostList>, div);
});

it('renders blog post list correctly', () => {
    render(<BlogPostList 
        postsWithComments={postsWithComments}
        fetchMoreComments={fetchMoreComments}
    ></BlogPostList>)
})

it('matches snapshot', () => {
    const tree = renderer.create(<BlogPostList 
        postsWithComments={postsWithComments}
        fetchMoreComments={fetchMoreComments}
    ></BlogPostList>).toJSON();

    expect(tree).toMatchSnapshot();
})