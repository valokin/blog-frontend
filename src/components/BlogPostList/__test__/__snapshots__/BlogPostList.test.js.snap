// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <h1>
    Blog
  </h1>
  <div
    className="infinite-scroll-component__outerdiv"
    style={Object {}}
  >
    <div
      className="infinite-scroll-component "
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "height": "auto",
          "overflow": "auto",
        }
      }
    >
      <div
        className="container"
        data-testid="blog-post-list-item"
        onClick={[Function]}
      >
        <div
          className="title"
          data-testid="blog-post-list-item-title"
        >
          title 1
        </div>
        <div
          className="description"
          data-testid="blog-post-list-item-description"
        >
          description 1
        </div>
        <div
          className="commentCount"
          data-testid="blog-post-list-item-comment-count"
        >
          Number of comments: 3
        </div>
      </div>
      <div
        className="container"
        data-testid="blog-post-list-item"
        onClick={[Function]}
      >
        <div
          className="title"
          data-testid="blog-post-list-item-title"
        >
          title 2
        </div>
        <div
          className="description"
          data-testid="blog-post-list-item-description"
        >
          description 2
        </div>
        <div
          className="commentCount"
          data-testid="blog-post-list-item-comment-count"
        >
          Number of comments: 3
        </div>
      </div>
      <div
        className="container"
        data-testid="blog-post-list-item"
        onClick={[Function]}
      >
        <div
          className="title"
          data-testid="blog-post-list-item-title"
        >
          title 3
        </div>
        <div
          className="description"
          data-testid="blog-post-list-item-description"
        >
          description 3
        </div>
        <div
          className="commentCount"
          data-testid="blog-post-list-item-comment-count"
        >
          Number of comments: 3
        </div>
      </div>
      <div
        className="container"
        data-testid="blog-post-list-item"
        onClick={[Function]}
      >
        <div
          className="title"
          data-testid="blog-post-list-item-title"
        >
          title 4
        </div>
        <div
          className="description"
          data-testid="blog-post-list-item-description"
        >
          description 4
        </div>
        <div
          className="commentCount"
          data-testid="blog-post-list-item-comment-count"
        >
          Number of comments: 3
        </div>
      </div>
      <div
        className="container"
        data-testid="blog-post-list-item"
        onClick={[Function]}
      >
        <div
          className="title"
          data-testid="blog-post-list-item-title"
        >
          title 5
        </div>
        <div
          className="description"
          data-testid="blog-post-list-item-description"
        >
          description 5
        </div>
        <div
          className="commentCount"
          data-testid="blog-post-list-item-comment-count"
        >
          Number of comments: 3
        </div>
      </div>
      <div
        className="container"
        data-testid="blog-post-list-item"
        onClick={[Function]}
      >
        <div
          className="title"
          data-testid="blog-post-list-item-title"
        >
          title 6
        </div>
        <div
          className="description"
          data-testid="blog-post-list-item-description"
        >
          description 6
        </div>
        <div
          className="commentCount"
          data-testid="blog-post-list-item-comment-count"
        >
          Number of comments: 3
        </div>
      </div>
    </div>
  </div>
</div>
`;
